### PR TITLE
Dogfood test-retry-plugin

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

This closes https://github.com/gradle/gradle-private/issues/2882 

So we can have the flakiness data in build scan.

This is how it looks like: https://e.grdev.net/s/vvmstvoqvp6qe/tests/flaky

### CODE REVIEW

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt [hunk 1]
```diff
@@ -250,8 +252,12 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
             configureJvmForTest()
             addOsAsInputs()
 
-            doFirst {
-                if (BuildEnvironment.isCiServer) {
+            if (BuildEnvironment.isCiServer) {
+                retry {
```
Comment (wolfs) [inline buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt]: We need to make sure that retries are not configured for the performance test task. Or at least that the plugin doesn't try to retry failed tests there.

**Comment by jjohannes** [comment]
> I wonder if we could somehow continue to use our own tracking.

What exactly do you mean by this (the GH board?) and why do you think it is needed?
I think the tracking so far never worked very well. While we have issues to work with, there are so many that we loose track. Maybe we should just start using GE and see where it gets us. We can still revive something else later.
In any case we have to decide how to work with the new tracking. For example, someone should regularly check the GE dashboard and create issues to actually work on fixing the most frequent tests etc.

> The ci health plugin can be removed I suppose.

:+1:

> I think we need to remove the failure condition to fail on failing tests or mark the build as successful

I think if the plugin is applied the flaky tests are not reported as failing (see the build scan linked in the description).

**Comment by wolfs** [comment]
> > I wonder if we could somehow continue to use our own tracking.
> 
> What exactly do you mean by this (the GH board?) and why do you think it is needed?
> I think the tracking so far never worked very well. While we have issues to work with, there are so many that we loose track. Maybe we should just start using GE and see where it gets us. We can still revive something else later.
> In any case we have to decide how to work with the new tracking. For example, someone should regularly check the GE dashboard and create issues to actually work on fixing the most frequent tests etc.

We can try the new tracking. We would look at the most flaky tests once a week maybe and then decide if there is something to do? And create issues for the tests with links to GE if we work on them?

> > I think we need to remove the failure condition to fail on failing tests or mark the build as successful
> 
> I think if the plugin is applied the flaky tests are not reported as failing (see the build scan linked in the description).

They are reported as failing: https://github.com/gradle/test-retry-gradle-plugin#teamcity

Only the exit code of Gradle changes.

**Comment by wolfs** [comment]
This is what currently happens to the performance tests: https://e.grdev.net/s/wttnpz52ftwfe/tests/failed?expanded=WzcsNF0

We skip all the tests, though the plugin seems to try to retry two of them...

**Comment by facewindu** [comment]
Regarding keeping the 'old tracking', I think switching to a pure GE-based approach allows to shake out any UX troubles you have with the tests dashboard, as a consumer.
This can help introducing features to it, if we realize that it misses stuff that real users would need.

**Comment by blindpirate** [comment]
> The ci health plugin can be removed I suppose.

I kept ci-health plugin because we still have a `TAG_BUILD` step. But after rethinking, I think this step should also be removed so we can completely use GE features to "shake out any UX troubles".

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt (hunk 1 above)
Comment (blindpirate) [reply buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt]: Done.

File: buildSrc/subprojects/plugins/plugins.gradle.kts [hunk 2]
```diff
@@ -11,6 +11,7 @@ dependencies {
     implementation("org.ow2.asm:asm:7.1")
     implementation("org.ow2.asm:asm-commons:7.1")
     implementation("com.google.code.gson:gson:2.7")
+    implementation("org.gradle:test-retry-gradle-plugin:1.1.0")
```
Comment (wolfs) [inline buildSrc/subprojects/plugins/plugins.gradle.kts]:
```suggestion
    implementation("org.gradle:test-retry-gradle-plugin:1.1.1")
```

Can you use the newer version, which fixes a bug which we also may hit?

**Comment by blindpirate** [comment]
@wolfs `1.1.1` used. Yes, the build steps come from the branch, not `master`.

**Comment by wolfs** [comment]
@blindpirate You'll need to accept the new plugin version for dependency verification.

**Pull request merged**